### PR TITLE
Refactor out CoffeScript ☕️🚫 

### LINKED ISSUE #25: Rewrite using ES6 and Standard

### PR DESCRIPTION
**Pull request by kutyel**
Here is the promised refactor, apparently we need more unit tests and I screw up something in the `Default.js` but maybe we can spot it together! 😎 

cc: @Kikobeats 

Closes #25

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,2 +1 @@
-require('coffee-script/register')
```
Comment (Kikobeats) [inline index.js:1]: bye my old friend

File: lib/Acho.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  const acho = Object.assign({ diff: [] }, DEFAULT(), opts)
+
+  acho.messages = (() =>
+    Object.keys(acho.types).reduce((messages, type) => {
+      acho[type] = acho.generateTypeMessage(type)
+      messages[type] = (opts.messages || {})[type] || []
+      return messages
+    }, {}))()
+
+  acho.push = function (type, ...messages) {
```
Comment (Kikobeats) [inline lib/Acho.js:17]: why not fat arrow for bind the context?

File: lib/Default.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+const ms = require('pretty-ms')
+
+const { colorize } = require('./Util')
+const CONST = require('./Constants')
+const format = require('./Format')
+
+module.exports = () => ({
+  print () {
+    return Object.keys(this.types).map(type =>
```
Comment (Kikobeats) [inline lib/Default.js:11]: fat arrow avoid return

File: lib/Acho.js (hunk 2 above)
Comment (kutyel) [reply lib/Acho.js:17]: Because wherever you need to use `this` you need to use a regular function, arrows don't work here

File: lib/Acho.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+'use strict'
+
+const CONST = require('./Constants')
+const DEFAULT = require('./Default')
+
+function Acho (opts = {}) {
```
Comment (kutyel) [inline lib/Acho.js:6]: Happened the same here, regular arrow functions don't have this so this logic breaks if it is not a regular function...

File: lib/Default.js (hunk 3 above)
Comment (kutyel) [reply lib/Default.js:11]: Same here, access to `this` makes it impossible to use arrows...

**Comment by kutyel** [comment]
Acho cómo va esto? 🤣

**Comment by Kikobeats** [comment]
Thanks for the effor migration the current implementation but tests are broken. Need to invest time trying to resolve them 😢

**Comment by Kikobeats** [comment]
Close in favour of https://github.com/achojs/acho/pull/48

**Pull request closed without merging**